wp-includes/widgets/class-wp-widget-media.php

```diff
@@ -420,4 +402,17 @@ public function render_control_template_scripts() {
 		</script>
 		<?php
 	}
+
+	/**
+	 * Whether the widget has content to show.
+	 *
+	 * @since 4.8.0
+	 * @access protected
+	 *
+	 * @param array $instance Widget instance props.
+	 * @return bool
+	 */
+	protected function has_content( $instance ) {
+		return ( $instance['attachment_id'] && 'attachment' !== get_post_type( $instance['attachment_id'] ) ) || $instance['url'];
```

Comment:
Not sure I understand the logic in this. Shouldn't it include `'attachment' === get_post_type( $instance['attachment_id'] )` (`===` not `!==`)?